src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -785,6 +801,40 @@ private void closeTempFile() {
     }
   }
 
+  private void fetchSchemaFromLatestFile() {
+    if (compatibility != StorageSchemaCompatibility.NONE && offset != -1) {
+      log.debug("Current schema is not set for partition {}, "
+          + "getting schema from file with latest offsets.", tp);
+      // check on WAL restored latest file to avoid recursive search
+      if (recovered && latestOffsetsFile != null) {
+        currentSchema = schemaFileReader.getSchema(
```

Comment:
It would produce the same result as the approach with the max offset file, the `latestOffsetsFile` path that is passed to it is the same filepath that the max offsets file would have. This happens after `wal.apply()` so the filesystem and WAL file are in sync at this point.

Reply:
So falling back to the search method would also produce the same exception.